client/my-sites/plans/jetpack-plans/product-lightbox/index.tsx

```diff
@@ -45,14 +47,39 @@ const ProductLightbox: React.FC< Props > = ( {
 	siteId,
 } ) => {
 	const close = useCallback( () => onClose?.(), [ onClose ] );
+	const dispatch = useDispatch();
 	const translate = useTranslate();
 
 	const onChangeOption = useCallback(
-		( productSlug: string ) => onChangeProduct( slugToSelectorProduct( productSlug ) ),
-		[ onChangeProduct ]
+		( productSlug: string ) => {
+			onChangeProduct( slugToSelectorProduct( productSlug ) );
+
+			// Tracking when variant selected inside the lightbox
+			dispatch(
+				recordTracksEvent( 'calypso_product_lightbox_variant_select', {
+					site_id: siteId,
+					product_slug: productSlug,
+				} )
+			);
+		},
+		[ onChangeProduct, dispatch, siteId ]
 	);
 
-	const { getCheckoutURL, getIsMultisiteCompatible, isMultisite } = useStoreItemInfoContext();
+	const { getCheckoutURL, getIsMultisiteCompatible, isMultisite, getOnClickPurchase } =
+		useStoreItemInfoContext();
+
+	const onClickCta = getOnClickPurchase( product );
+	const onCheckoutClick = () => {
+		onClickCta();
+
+		// Tracking when checkout is clicked
+		dispatch(
+			recordTracksEvent( 'calyspo_product_lightbox_checkout_click', {
+				site_id: siteId,
+				product_slug: product.productSlug,
+			} )
+		);
+	};
```

Comment:
Maybe wrap this in `useCallback`?
```suggestion
	const onCheckoutClick = useCallback( () => {
		getOnClickPurchase( product )();

		// Tracking when checkout is clicked
		dispatch(
			recordTracksEvent( 'calyspo_product_lightbox_checkout_click', {
				site_id: siteId,
				product_slug: product.productSlug,
			} )
		);
	}, [ dispatch, getOnClickPurchase, product, siteId ] );
```